conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,700 @@
+    io:println(err4.message() == "error message 4"); // @output true
+    io:println(err4.cause() is E); // @output true
+    io:println(err4.cause() is ()); // @output false
+    io:println(err4.cause() === err); // @output true
+}
+
+Test-Case: output
+Description: Test evaluating the error-constructor-expr constructs a new immutable detail mapping,
+             where each named-arg specifies a field of the error detail mapping and
+             the static type of each named-arg must be a subtype of value:Cloneable.
+             (Note: value:Cloneable requires explicit importing of lang.value. Hence, using error:Cloneable as they are equivalent)
+Labels: DecimalFloatingPointNumber, error-constructor-expr, error-type, error-type-parameter, error:Cloneable,
+        error:detail, intersection-type, is-expr, list-constructor-expr, map-type, member-access-expr, method-call-expr,
+        module-type-defn, readonly-type, record-type, string
+
+type E error<record {|string message; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("error message 1");
+    io:println(err1.detail() is map<error:Cloneable>); // @output true
+    io:println(err1.detail() is readonly); // @output true
+    io:println(err1.detail()); // @output {}
+
+    error err = error("ERROR!");
+    error err2 = error("error message 2", err);
+    io:println(err2.detail() is map<error:Cloneable>); // @output true
+    io:println(err2.detail() is readonly); // @output true
+    io:println(err2.detail()); // @output {}
+
+    map<string> m = {"x": "x", "y": "y"};
+    error err3 = error("error message 3", a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err3.detail() is map<error:Cloneable>); // @output true
+    io:println(err3.detail() is readonly); // @output true
+    io:println(err3.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err4 = error("error message 4", err, a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err4.detail() is map<error:Cloneable>); // @output true
+    io:println(err4.detail() is readonly); // @output true
+    io:println(err4.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err5 = error E("error message 1", message = "Oops!");
+    io:println(err5.detail() is map<error:Cloneable>); // @output true
+    io:println(err5.detail() is readonly); // @output true
+    io:println(err5.detail()); // @output {"message":"Oops!"}
+
+    error err6 = error E("error message 2", err, message = "Oops!", fatal = true);
+    io:println(err6.detail() is readonly); // @output true
+    io:println(err6.detail() is map<error:Cloneable>); // @output true
+    io:println(err6.detail()["message"] is error:Cloneable); // @output true
+    io:println(err6.detail()["message"] is readonly); // @output true
```

Comment:
Are these intended to test `with its members being the result of applying the ImmutableClone abstract operation to the result of evaluating each named-arg and every defaultable arg.`? Or are there other tests?

These don't quite test that since string and boolean are anyway immutable.

Reply:
The intention was the former reason. Removed as they are anyway immutable.